src/XamlX/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace System;
+
+/// <summary>
+/// Source https://stackoverflow.com/a/46382137/20894223
+/// </summary>
+internal static class TypeExtensions
+{
+    public static bool IsInternal(this Type t) => !t.IsVisible
```

Comment:
The method explicitly excludes nested internal types, so the naming is clearly wrong here.